conbench/app/_endpoint.py

```diff
@@ -11,8 +11,43 @@ def as_bool(x):
 
 class AppEndpoint(flask.views.MethodView):
     def public_data_off(self):
-        public = as_bool(os.getenv("BENCHMARKS_DATA_PUBLIC", "yes"))
-        return not flask_login.current_user.is_authenticated and not public
+        """
+        Compute whether data/view should be hidden from an anonymous user.
+
+        When BENCHMARKS_DATA_PUBLIC is set to a true-ish value then always show
+        benchmark data (regardless of whether the request is anonymous or not).
+
+        When BENCHMARK_DATA_PUBLIC is set to a false-ish value then decide to
+        hide the view (return True) when the access is from anonymous.
+
+        Note, just a quick thought dump: this needs a bit of consolidation for
+        easier code readability, and for de-duplicating per-view code. In
+        individual views this method is so far used to short-cut request
+        processing and redirect to the login page if needed. Maybe build a
+        simple decorator instead with a name that relates to access control
+        (such as `enforce_access_control()`). Decorated views enforce access
+        control (with the business logic as in _this_ function here),
+        non-decorated views do not enforce. If access is denied then the
+        redirect to the login page should be emitted from that decorator -- the
+        redirect will contain view-specific parameters (target URL).
+        """
+
+        # TODO(JP): the environment variable readout should be managed via the
```

Comment:
I like this idea; want to do it now?

Reply:
Thanks for having a look! That's part of my attempt to not do toooo much in a single pull request, so I will leave this for the follow-up work on access control:

> Follow-up work is required to introduce the desired UX: upon an authentication required error and the corresponding redirect to the login page -- that's where we will use that feature. That work will be done in subsequent patches.